src/browser/browser.zig

```diff
@@ -380,7 +364,37 @@ pub const Page = struct {
 
         var response = try request.sendSync(.{});
         const header = response.header;
-        try self.session.cookie_jar.populateFromResponse(self.uri, &header);
+        try self.session.cookie_jar.populateFromResponse(request.uri, &header);
+
+        // update uri after eventual redirection
+        var buf: std.ArrayListUnmanaged(u8) = .{};
+        defer buf.deinit(arena);
+
+        buf.clearRetainingCapacity();
+        try request.uri.writeToStream(.{
+            .scheme = true,
+            .authentication = true,
+            .authority = true,
+            .path = true,
+            .query = true,
+            .fragment = true,
+        }, buf.writer(arena));
+        self.rawuri = try buf.toOwnedSlice(arena);
+
+        self.uri = try std.Uri.parse(self.rawuri.?);
+
+        // TODO handle fragment in url.
+        self.url = try URL.constructor(arena, self.rawuri.?, null);
+        self.location.url = &self.url.?;
+        try self.session.window.replaceLocation(&self.location);
+
+        // prepare origin value.
+        buf.clearRetainingCapacity();
```

Comment:
As-above, if you stick with the `toOwnedSlice(arena)`, then this code does nothing. `toOwnedSlice(arena)` calls `clearAndFree()`.

If, above, you decide to do instead:

```
self.rawuri = buf.items
```

Then here you'll want to reset the buffer:

```
buf = .{};
```

as the `clearRetainingCapacity()` will break `self.rawuri`